scripts/install-macos.sh

```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
+# shellcheck disable=SC2016
+set -e
+
+# Set this machine's dependencies status(es) with the OK flag
+OK=true;
+
+# Dependency test functions begin here
+
+function isNode() {
+    if node -v; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isYarn() {
+    if yarn --version; then
+        OK=true;    
+    else
+        OK=false;
+    fi
+}
+
+function isGit() {
+    if git --version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isGo() {
+    if go version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isGlide() {
+    if glide --version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+# Installation functions begin here
+
+function installGo() {
+    echo 'Installing Golang on your machine.'
+    # Go install script based on https://github.com/canha/golang-tools-install-script
+    VERSION="1.15.7"
+
+    [ -z "$GOROOT" ] && GOROOT="$HOME/go/go-install-do-not-delete"
+    [ -z "$GOPATH" ] && GOPATH="$HOME/go"
+
+    OS="$(uname -s)"
+    ARCH="$(uname -m)"
+
+    case $OS in
+        "Linux")
+            case $ARCH in
+            "x86_64")
+                ARCH=amd64\
+                ;;
+            "aarch64")
+                ARCH=arm64
+                ;;
+            "armv6" | "armv7l")
+                ARCH=armv6l
+                ;;
+            "armv8")
+                ARCH=arm64
+                ;;
+            .*386.*)
+                ARCH=386
+                ;;
+            esac
+            PLATFORM="linux-$ARCH"
+        ;;
+        "Darwin")
+            PLATFORM="darwin-amd64"
+        ;;
+    esac
+
+    if [ -z "$PLATFORM" ]; then
+        echo "Your operating system is not supported by the script."
+        exit 1
+    fi
+
+    if [ '/bin/bash' == `which bash` ]; then 
+        shell_profile="$HOME/.bashrc"
+        echo "bash is installed"
+    else
+        echo "Kelp requires bash"
+    fi
+
+    {
```

Comment:
we can check if GOROOT is defined in the bashrc file before we do this -- so it doesn't repeat it in the bashrc file.

we can achieve this by doing a grep for "export GOROOT" on the $shell_profile and seeing if it retuned any result. you may need to do something like this to check that:

``if [ `grep "export GOROOT" $shell_profile | wc -l` -eq 0 ]; # if equals 0 then we should append``

we can reuse the condition check for `-d "$GOROOT"` that you already have below this append statement